mxcubecore/HardwareObjects/ANSTO/prefect_flows/prefect_client.py

```diff
@@ -71,6 +71,63 @@ async def trigger_flow(self, wait=False) -> FlowRunResponse:
         if wait:
             await self.wait()
 
+    async def trigger_data_collection(self, poll_interval: float = 3.0) -> None:
+        """
+        Triggers a prefect flow but only waits until data collection is finished
+        so that data processing happens in the background
+
+        Parameters
+        ----------
+        poll_interval : float, optional
+            The poll interval in seconds, by default 3.0 seconds
+
+        Returns
+        -------
+        None
+        """
+        self.deployment_id = await self.get_deployment_id_from_name(self.name)
+
+        response: FlowRunResponse = (
+            await self.prefect_client.create_flow_run_from_deployment(
+                self.deployment_id, parameters=self.parameters
+            )
+        )
+        self.flow_run_id = response.id
+
+        task_len = 0
+        while task_len == 0:
+            await asyncio.sleep(poll_interval)
+            tasks = await self.get_tasks()
+            task_len = len(tasks)
+
+        state = "RUNNING"
+        while state == "RUNNING":
+            await asyncio.sleep(poll_interval)
+            tasks = await self.get_tasks()
+            data_collection_task = tasks[0]
+            state = data_collection_task.state.type
+
+    async def get_tasks(self, flow_run_id: UUID = None) -> None:
+        """
+        Gets prefect tasks
+
+        Parameters
+        ----------
+        flow_run_id : UUID, optional
+            The flow id, by default None
+
+        Returns
+        -------
+        tasks
+            The tasks
+        """
```

Comment:
Looks like the type hint needs to be updated here, currently implies the method returns `None`.